wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/urlfactory/URLFactoryBuilder.java

```diff
@@ -135,9 +135,15 @@ public Optional<FactoryMetaDto> createFactoryFromDevfile(
       if (isNullOrEmpty(devfileYamlContent)) {
         return Optional.empty();
       }
-
       try {
         JsonNode parsedDevfile = devfileParser.parseYamlRaw(devfileYamlContent);
+        // We might have an html content in the parsed devfile, in case if the access is restricted,
+        // or if the URL points to a wrong resource.
+        try {
+          devfileVersionDetector.devfileVersion(parsedDevfile);
+        } catch (DevfileException e) {
+          throw new ApiException("Failed to fetch devfile");
```

Comment:
?
```suggestion
          throw new ApiException("Devfile is not valid");
```

Reply:
It is not true to say that the devfile is not valid because it is not a devfile. It is an HTML content of a repository, or an authentication page in case if the repo is private.

Reply:
Since this is going to be exposed to the user we may want to provide more information:
- What happened: "We have tried to download a Devfile but the content is not YAML."
- Why it happened: "The provided URL directed to HTML content or to an authentication page."
- Possible workaround: "Fix this problem providing a raw Devfile URL or configuring authentication for the git service provider".

Reply:
@vinokurig agree with Mario, I believe we need to provide more details

Reply:
This is supposed to be handled by the dashboard side, see: https://github.com/eclipse-che/che-dashboard/pull/754